### PR TITLE
Electriclighting bug fixes

### PR DESCRIPTION
**Pull request by sariths**
Addressed the bugs in https://github.com/mostaphaRoudsari/Honeybee/issues/556 except adding the Type B photometry. That will take some time as some changes to coordinate system of the luminaire might have to be made in order to make it work properly.

(@mostaphaRoudsari I haven't committed anything to this library in a long time. Can you please check if my PR is inorder before merging it.)

### CODE REVIEW

**Comment by chriswmackey** [comment]
@sariths ,
Looks great and I'm merging it in now!
-Chris

File: src/Honeybee_IES Luminaire.py [hunk 1]
```diff
@@ -816,7 +820,7 @@ def repSpc(originalString):
 
     if _writeRad:  
         if len(_iesFilePath)==1:
-            _iesFilePath = filePath
+            _iesFilePath = _iesFilePath[0]
```
Comment (mostaphaRoudsari) [inline src/Honeybee_IES Luminaire.py]: Apparently _iesFilePath, = _iesFilePath is faster than _iesFilePath, = _iesFilePath[0]
This is something the I learned recently but it is not really readable. Just sharing.
source: https://twitter.com/raymondh/status/773786851727380480

**Comment by sariths** [comment]
9-11-2016 6-24-27 pm

Interesting, but I couldn't get it to work.. Is that a python 3 thing?

**Comment by mostaphaRoudsari** [comment]
It only works for a list with a single item as it says in the tweet. Try this:
```python
a = range(1)
a, = a
print a
```
This ValueError can be quite useful in cases such as yours. Your code will look like this:

```python
        try:
            _iesFilePath, = _iesFilePath[0]
        except ValueError:
            newIesFile = os.path.join(dirpath,filenamefull+".ies")
            with open(newIesFile,'w') as iesFileWrite:
                for lines in _iesFilePath:
                    print(lines,file=iesFileWrite,end="\n")
            _iesFilePath = os.path.join(dirpath,filenamefull+".ies")
       else:
            if not os.path.exists(_iesFilePath):
                newIesFile = os.path.join(dirpath,filenamefull+".ies")
                with open(newIesFile,'w') as iesFileWrite:
                    print(_iesFilePath,file=iesFileWrite,end="\n")
                _iesFilePath = os.path.join(dirpath,filenamefull+".ies")

```

**Comment by sariths** [comment]
Aha!...overlooked the length aspect of it! Makes sense now...

**Comment by chriswmackey** [comment]
Very good to know, @mostaphaRoudsari .  Thanks for the tip.

**Pull request merged**